.github/workflows/codeql-analysis.yml

```diff
@@ -66,5 +67,15 @@ jobs:
     #   make bootstrap
     #   make release
 
+    - name: Set up JDK ${{ matrix.java }}
+      uses: actions/setup-java@0ab4596768b603586c0de567f2430c30f5b0d2b0 # v3.13.0
```

Comment:
Why this specific commit? Why not version 4.0? 

This is going to have to change at some point. Is there an issue to track 'update setup-java to <version>'?

Reply:
Should have been 4.0.0 (I've updated that). GH recommends pinning versions to the hash commmit as (something like) v4 is effectively an alias that can silently be changed under the hood. Dependendabot was updated to deal with the hashcommit version numbers and accompanying comment (it will update both)